dapps/profile.vue/src/components/did/Delegates.ts

```diff
@@ -0,0 +1,109 @@
+/*
+  Copyright (C) 2018-present evan GmbH.
+
+  This program is free software: you can redistribute it and/or modify it
+  under the terms of the GNU Affero General Public License, version 3,
+  as published by the Free Software Foundation.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see http://www.gnu.org/licenses/ or
+  write to the Free Software Foundation, Inc., 51 Franklin Street,
+  Fifth Floor, Boston, MA, 02110-1301 USA, or download the license from
+  the following URL: https://evan.network/license/
+*/
+
+import Component, { mixins } from 'vue-class-component';
+import { EvanComponent, EvanTableColumn, ContactInterface } from '@evan.network/ui-vue-core';
+import { bccUtils } from '@evan.network/ui';
+import { isEqual } from 'lodash';
+import { Prop } from 'vue-property-decorator';
+import { DidService } from './DidService';
+
+interface Delegate {
+  did: string;
+  note: string;
+}
+@Component
+export default class DelegatesComponent extends mixins(EvanComponent) {
+  @Prop() delegates: Delegate[];
+
+  isEditMode = false;
+
+  contacts: ContactInterface[] = [];
+
+  columns: EvanTableColumn[] = [
+    {
+      key: 'did',
+      label: this.$t('_profile.did.did'),
+    },
+    {
+      key: 'note',
+      label: this.$t('_profile.did.note'),
+    },
+    {
+      key: 'action',
+      label: '',
+      thClass: 'th-icon',
+    },
+  ]
+
+
+  previousData: Delegate[] = [];
+
+  async mounted(): Promise<void> {
+    this.contacts = await bccUtils.getContacts(this.getRuntime());
+  }
+
+  /**
+   * Temporarily add contact to delegates
+   * @param contact selected contact
+   */
+  async onSelectContact(contact: ContactInterface): Promise<void> {
+    this.delegates = [...this.delegates, {
+      did: contact.value,
+      note: contact.label,
+    }];
+  }
+
+  saveDelegates(): void {
+    DidService.saveDelegates(this.delegates);
+    this.isEditMode = false;
+  }
+
+  /**
+   * Removes the selected delegate temporarily
+   * @param index row index of the item to be removed
+   */
+  deleteDelegate(index: number): void {
+    this.delegates = this.delegates.filter((_, i) => i !== index);
```

Comment:
should also work:

```suggestion
    this.delegates.splice(index, 1);
```

Reply:
I always prefer the non-mutating approach where applicable.

Reply:
bc the haschanges function will  not work otherwise? 😉

Reply:
Uhm, I'm not quite sure what you mean with hashchanges? 😅